rfc/rfc-0000-k8s-events.md

```diff
@@ -0,0 +1,170 @@
+key guiding principles:
+
+1. > Let it be something that isn't satisfactorily presented in other user views (eg: `k get workload`).
+2. > Let the event list be as meaningful as is necessary, and no more.
+
+The cartographer controller could emit the following events:
+
+| Reason | Message Format | Description | involvedObject |
+| --- | --- | --- | --- |
+| ResourceExternalSpecChange | an external actor changed the spec of `<group>.<kind>/<name>` | Our cache of the spec, and the spec we just generated match, but the API server has a different one. A lot of these is evidence of thrashing with external resources | Owner |
+| ResourceExternalSpecChange | an external actor changed the spec of resource: `<resource name>` for object `<group>.<kind>` | Our cache of the spec, and the spec we just generated match, but the API server has a different one. A lot of these is evidence of thrashing with external resources | Blueprint |
+| ResourceGetError | `<resource name>` could not retrieve `<group>.<kind>/<name>` due to error: `<error message>` | Loading a resource is failing due to a client.get issue (missing is not an error) | Owner |
+| ResourceGetError | `<resource name>` could not retrieve `<group>.<kind>` due to error: `<error message>` | Loading a resources is failing due to a client.get issue (missing is not an error) - this could be spammy | Blueprint |
+| ResourceGarbageCollected | `<group>.<kind>/<name>` is no longer referenced | This owner has selected a different template, either by supply chain selection or templating, and this object is no longer needed | Owner |
+| ImmutableResourceGarbageCollected | `<n> * <group>.<kind>` historical objects deleted due to garbage collection policy | This runnable's GC policy has caused `n` objects to be removed | Runnable |
+| ResourceInvalid | `<resource name>` could not be applied as `<group>.<kind>/<name>` due to API server error `<error message>` | This object was (probably) malformed. | Owner |
+| ResourceInvalid | `<resource name>` could not be applied as `<group>.<kind>` due to API server error `<error message>` | This object was (probably) malformed. This lets operators know their templates might have issues | Blueprint |
+| ResourceDoesNotExist | `<resource name>` could not be applied because `<group>.<kind>/<name>` does not exist on this cluster | Did someone forget to install the CRDs? Otherwise it's a malformed template | Owner |
+| ResourceDoesNotExist | `<resource name>` could not be applied because `<group>.<kind>` does not exist on this cluster | Did someone forget to install the CRDs? Otherwise it's a malformed template | Blueprint |
+| ResourceApplied | `<resource name>` was applied as `<group>.<kind>/<name>` | a resource needed to be created/updated | Owner |
+| ResourceKindChanged | `<resource name>` was `<old group>.<old kind>`, now `<new group>.<new kind>` | YTT selection for a template kind changed | Owner |
```

Comment:
nit:

as scott pointed out in a previous RFC, this could happen in regular templating too, as the `apiVersion` and `kind` could be templated as well.

Reply:
I missed that, good point.